db-client-java/src/main/java/com/eventstore/dbclient/ReadSubscription.java

```diff
@@ -64,10 +59,31 @@ public void request(long n) {
         if (n <= 0) {
             subscriber.onError(new IllegalArgumentException("non-positive subscription request: " + n));
         }
-        lock.lock();
-        requested.updateAndGet(current -> current + n);
-        hasRequested.signal();
-        lock.unlock();
+
+        request0(n);
+    }
+
+    private void request0(long n)
```

Comment:
The only it's called that is because the public method does arg check, and this does not, so they are closely related. Usually in other codebases I have seen this naming convention, but feel free to merge the PR and then rename it. All of the PR should be taken as just a starting point, not the final answer.